### PR TITLE
feat: Implement GraphQL Schema Diagram with Relationships 

### PR DESCRIPTION
**Pull request by dthyresson**
A start at making a GraphQL schema diagram with ReactFlow using Custom Nodes.

This is a simple diagram of your main sdl mode/types. In doesn't o inputs, scalars, etc. Just models and queries and mutations so you get a quick digestible view of the main part of the schema,

The rest of the info is available in the Playground explorer

### Diagram

image

### Relationships Table

image

### Future

Future PR's will need  to

* layout : see?: https://reactflow.dev/learn/layouting/layouting#dagre
* show fields?

### CODE REVIEW

**Comment by dthyresson** [comment]
Add happy with this as a starting point. Want to move on to Prisma schema ERD viz tomorrow.

More to do with layout and some field info, but core features present.

File: api/src/services/graphqlInspector/graphqlInspector.ts [hunk 1]
```diff
@@ -1,12 +1,42 @@
-import { coverage } from '@graphql-inspector/core'
+import { validate } from '@graphql-inspector/core'
```
Comment (dthyresson) [inline api/src/services/graphqlInspector/graphqlInspector.ts:1]: I'm not sure I'll keep all this functionality.

File: api/src/services/graphqlInspector/graphqlInspector.ts [hunk 2]
```diff
@@ -1,12 +1,42 @@
...
 
-// import { makeMergedSchema } from "@redwoodjs/graphql-server"
+import { logger } from 'src/lib/logger'
+import { getSchema } from 'src/util/graphqlSchema'
+
+/**
+ *
+ * @returns import {
+  diff,
+  validate,
```
Comment (Tobbe) [inline api/src/services/graphqlInspector/graphqlInspector.ts:12]: This looks strange :)  It started out as a doc comment, but then turned into something else?

File: api/src/util/graphqlSchema.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+import { getCachedDocumentNodeFromSchema } from '@graphql-codegen/plugin-helpers'
+import { CodeFileLoader } from '@graphql-tools/code-file-loader'
+import { loadSchema, LoadSchemaOptions } from '@graphql-tools/load'
+import { print, visit } from 'graphql'
+
+import { rootSchema } from '@redwoodjs/graphql-server'
+import { getConfigPath, getPaths } from '@redwoodjs/project-config'
+
+// import { logger } from 'src/lib/logger'
```
Comment (Tobbe) [inline api/src/util/graphqlSchema.ts:12]: Remove this?

File: api/src/util/graphqlSchema.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+// this seems a wrong way to import this type?
```
Comment (Tobbe) [inline api/src/util/graphqlSchema.ts:14]: Why? What's wrong?

File: api/src/util/graphqlSchema.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+export const getGraphQLSchemaInfo = async () => {
+  const schema = await getSchema()
+  const documentNode = getCachedDocumentNodeFromSchema(schema)
+
+  const ast = JSON.stringify(documentNode.definitions)
+  const id = crypto.createHash('md5').update(ast).digest('hex')
+
+  const definitions = []
+
+  // logger.info({ schema }, 'GraphQL schema')
```
Comment (Tobbe) [inline api/src/util/graphqlSchema.ts:62]: Remove?

File: api/src/util/graphqlSchema.ts [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+      definitions.push(node)
+    },
+    OperationDefinition: (node) => {
+      definitions.push(node)
+    },
+  })
+
+  const relationships = getGraphQLRelationships(definitions)
+
+  // logger.info({ relationships }, 'GraphQL Relationships')
```
Comment (Tobbe) [inline api/src/util/graphqlSchema.ts:84]: More commented code

File: web/src/Routes.tsx [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import { Router, Route, Set } from '@redwoodjs/router'
 
 import SidebarLayout from 'src/layouts/SidebarLayout'
+import GraphQLSchemaPage from 'src/pages/GraphQLSchemaPage/GraphQLSchemaPage'
```
Comment (Tobbe) [inline web/src/Routes.tsx:4]: Remove please. Are VSCode not picking up on the new pages you create? Are you running with `yarn rw dev`?

File: api/src/services/graphqlInspector/graphqlInspector.ts (hunk 1 above)
Comment (dthyresson) [reply api/src/services/graphqlInspector/graphqlInspector.ts]: Decided to remove functionally but got service to run converse without errors.

There are comments about next steps -- properly loading the source documents to apply coverage,

I tried a few methods (graphql-tools code file loader, glob filenames, etc) but I'll check with Kamal from the Guild actually how to do it. It expects a GraphQL "Source" which I thought was the last, but didn't quite work.

File: api/src/services/graphqlInspector/graphqlInspector.ts (hunk 2 above)
Comment (dthyresson) [reply api/src/services/graphqlInspector/graphqlInspector.ts]: Removed. Better comment now.

File: api/src/util/graphqlSchema.ts (hunk 4 above)
Comment (dthyresson) [reply api/src/util/graphqlSchema.ts:14]: It's at the same level as src. For a second I was confused about api/src/types and api/types. I need api/types.

**Pull request merged**